datafusion/core/tests/sql/window.rs

```diff
@@ -471,3 +471,297 @@ async fn window_with_agg_in_expression() -> Result<()> {
     assert_batches_eq!(expected, &actual);
     Ok(())
 }
+
+#[tokio::test]
+async fn window_frame_empty() -> Result<()> {
+    let ctx = SessionContext::new();
+    register_aggregate_csv(&ctx).await?;
+    let sql = "SELECT \
+               SUM(c3) OVER(),\
+               COUNT(*) OVER ()\
+               FROM aggregate_test_100 \
+               ORDER BY c9 \
+               LIMIT 5";
+    let actual = execute_to_batches(&ctx, sql).await;
+    let expected = vec![
+        "+----------------------------+-----------------+",
+        "| SUM(aggregate_test_100.c3) | COUNT(UInt8(1)) |",
+        "+----------------------------+-----------------+",
+        "| 781                        | 100             |",
+        "| 781                        | 100             |",
+        "| 781                        | 100             |",
+        "| 781                        | 100             |",
+        "| 781                        | 100             |",
+        "+----------------------------+-----------------+",
+    ];
+    assert_batches_eq!(expected, &actual);
+    Ok(())
+}
+
+#[tokio::test]
+async fn window_frame_rows_preceding() -> Result<()> {
+    let ctx = SessionContext::new();
+    register_aggregate_csv(&ctx).await?;
+    let sql = "SELECT \
+               SUM(c4) OVER(ROWS BETWEEN 1 PRECEDING AND 1 FOLLOWING),\
```

Comment:
I am surprised this query has deterministic results -- I would expect the OVER clause needs an ORDER BY a well (otherwise the values that c4 will be computed could be different.

Something like:

```suggestion
               SUM(c4) OVER(ORDER BY c4 ROWS BETWEEN 1 PRECEDING AND 1 FOLLOWING),\
```
(and the same for the OVER below as well)

Reply:
I believe the `ORDER BY c9` below makes this deterministic. Except when there are duplicates in `c9` with different values in `c4`, as you also note below. In that case it would probably be best to extend the `ORDER BY` with another column that makes it stable.